### DISCUSSION TITLE
(arm64) Turned into Custom Path segment

### DISCUSSION THREAD

**Discussion by sakabug** [JanDeDobbeleer/oh-my-posh#888] [General]
I have currently oh-my-posh and posh-git installed in Win10 machine. They work all fine and tandy, but I'd like them to work in my home server connection too.

Do I need to install the oh-my-posh on to my Ubuntu Server too? I'm running Ubuntu Server on RasPi 4B as 64bit, so would I need arm64 version?

**Comment by JanDeDobbeleer** [comment]
Yes, indeed. That should work.

**Reply by sakabug** [reply]
By the way. I had earlier some old v2 oh-my-posh and was able to do this:  [Imgur link](https://i.imgur.com/NKMmIKk.jpg)

So basically path format on long cases show folder and parentfolder, handled in function at profile.ps1 with few if else cases.
Is this still achievable some way in V3?

**Reply by JanDeDobbeleer** [reply]
Can you give some examples, if not we can always extend the functionality.

**Reply by sakabug** [reply]
On v2 we could edit that path in powershell profile.ps1 like this:

```
function miniPath(){
    $dirSep = [IO.Path]::DirectorySeparatorChar
    $pathComponents = $PWD.Path.Split($dirSep)
    $displayPath = if ($pathComponents.Count -gt 3) {
        '..{0}{1}' -f $dirSep, ($pathComponents[-2,-1] -join $dirSep)
    } elseif ($PWD.Path.Length -gt 3){
    $PWD.Path
    }	else {
	$PWD.Path.Substring(0,$PWD.Path.Length-1)
	}
	"{0} " -f $displayPath
}
```
And the miniPath was probably called in some V2 file which I remember editing.
(Now, that I looked I didn't have the drive letter set into the path, but not too relevant.)

So I think in V3 we can't chop the path into pieces ourselves if I'm right.

**But** if you were to add in block Path's style **agnoster** so that you can change it's **threshold** just like in **mixed**, end result would be more satisfactory.

For example: Now if we have **agnoster**
x:\\..\\..\\..\\..\multibutton

If there would be **threshold** for the depth that is show, we could get:
x:\\..\\..\\..\examples\multibutton

...And before I start sounding some entitled prick who's not happy. I love oh-my-posh 👍🏻

**Reply by JanDeDobbeleer** [reply]
I like the threshold for the path dots tbh. That could work.

**Reply by sakabug** [reply]
Just figured out:
[
Link to Imgur](https://i.imgur.com/GBbB21v.jpg)

'changePOSH 2 $true 1' here gives me 2 folder names and 1 letter from the rest of the folders.
Of course this example applies only for PowerShell.

**Reply by sakabug** [reply]
- drive on/off
- full name count
- partial name count
- partial letter count
- excluded string
- partial postfix
- custom separator
![oh-my-posh](https://user-images.githubusercontent.com/79233780/131389384-c0acc570-f5fe-4d1a-b91d-d0db2838e0d0.png)

https://i.imgur.com/Iw9sYah.png
